test/test_lib/utils.js

```diff
@@ -166,16 +166,24 @@ Utils.prototype = {
     assert(false, "Did not revert as expected.");
   },
 
-  /// @dev Expect failure from assert, but returns error instead
-  expectFailedAssert: async (promise) => {
+  /**
+   * Asserts that a given ethereum call/transaction leads to a assert failure.
+   * The call/transaction is given as a promise.
+   *
+   * @param {promise} promise Awaiting this promise must lead to a error.
+   * @param {string} expectedMessage If given, the returned error message must
+   *                                 include this string (optional).
+   */
+  expectFailedAssert: async (promise, expectedMessage) => {
     try {
       await promise;
     } catch (error) {
       assert(
-        error.message.search('invalid opcode') > -1,
+        error.message.search('Returned error:') > -1,
         'The contract should fail an assert. Instead: ' + error.message
       );
 
+      assertExpectedMessage(expectedMessage, error);
```

Comment:
What kind of message would you expect with an assert? An assert would always return `invalid opcode`, or not?

Reply:
`invalid opcode` is one of the assertion failures.
We are also interested in the `sender doesn't have enough funds to send tx.` error

Reply:
👍